slides/aml-10-021918-trees-forests/index.html

```diff
@@ -0,0 +1,724 @@
+   --
+   .center[![:scale 50%](images/img_6.png)]
+
+
+ 
+ ![:scale 5%](images/img_7.png) observations in node m
+ 
+ ![:scale 3%](images/img_8.png) classes
+ 
+ ![:scale 5%](images/img_9.png) distribution over classes in node m
+???
+---
+
+# Prediction
+
+.center[
+![:scale 80%](images/img_10.png)
+]
+
+ - Traverse tree based on feature tests
+ - Predict most common class in leaf
+
+???
+
+---
+
+# Regression trees
+
+ - Impurity Criteria:
+ - Mean Squared Error
+ - Mean Absolute Error
+ 
+ 
+ - Prediction:
+ - Predict mean.
+ 
+
+ - Without regularization / pruning:
+ - Each leaf often contains a single point to be “pure”
+???
+
+---
+
+# Visualizing trees with sklearn
+
+.smaller[
+```python
+from sklearn.datasets import load_breast_cancer
+cancer = load_breast_cancer()
+```
+```python
+X_train, X_test, y_train, y_test = train_test_split(cancer.data,canc
+er.target, stratify=cancer.target,random_state=0)
+```
+```python
+#tree visualization
+```
+```python
+from sklearn.tree import DecisionTreeClassifier, export_graphviz
+tree = DecisionTreeClassifier(max_depth=2)
+tree.fit(X_train, y_train) 
+```
+```python
+tree_dot = export_graphviz(tree, out_file=None, feature_names=cancer.feature_names)
+print(tree_dot)
+```
+]
+
+???
+---
+# Visualizing trees with sklearn
+.center[
+![:scale 80%](images/img_11.png)
+]
+???
+---
+
+
+# Showing dot files in Jupyter
+ - First install graphvis C library:
+   conda install graphviz
+ - Then install graphviz python library:
+ - pip install graphviz
+
+ .center[
+ ![:scale 60%](images/img_12.png)
+ ]
+
+
+???
+---
+class:spacious
+
+# Parameter Tuning
+
+ - Pre-pruning and post-pruning (not in sklearn yet)
+ - Limit tree size (pick one):
+   max_depth
+   --
+   max_leaf_nodes
+   --
+   min_samples_split
+   --
+   (and more)
+???
+---
+class:spacious
+
+# No pruning
+.center[
+![:scale 100%](images/img_13.png)
+]
+???
+
+---
+
+# max_depth = 4
+
+.center[
+![:scale 100%](images/img_14.png)
+]
+
+???
+---
+
+# max_leaf_nodes = 8
+.center[
+![:scale 50%](images/img_15.png)
+]
+
+???
+---
+
+# min_samples_split = 50
+.center[
+![:scale 70%](images/img_16.png)
+]
+
+???
+---
+
+.smaller[
+```python
+from sklearn.model_selection import GridSearchCV
+param_grid = {'max_depth':range(1, 7)}
+grid = GridSearchCV(DecisionTreeClassifier(random_state=0),param_grid=param_grid,
+cv=10)
+grid.fit(X_train, y_train)
+```
+]
+.center[
+![:scale 70%](images/img_17.png)
+]
+
+???
+---
+.smaller[
+```python
+from sklearn.model_selection import GridSearchCV
+param_grid = {'max_leaf_nodes':range(2, 20)}
+grid = GridSearchCV(DecisionTreeClassifier(random_state=0),
+param_grid=param_grid, cv=10)
+grid.fit(X_train, y_train)
+```
+]
+.center[
+![:scale 70%](images/img_18.png)
+]
+
+???
+---
+.center[
+![:scale 95%](images/img_19.png)
+]
+
+???
+---
+.center[
+![:scale 95%](images/img_20.png)
+]
+???
+---
+#Extrapolation
+.center[
+![:scale 85%](images/img_20.png)
+]
+???
+---
+
+class: split-40
+# Instability
+
+.column[
+.smaller[
```

Comment:
maybe see if you can make it even smaller (I had a "smallest" class somewhere)? right now it's impossible to read :-/

Reply:
I didnt find a smallest class so I added another class called tiny code which has made this smaller. Hope it looks better now. Will create a new pull request with all these changes in a little while for you to review.